File: java/digital_root.java
public class DRoot {
	public static int digital_root(int n) {

		while (n > 9) {
			int sum = 0;
			String stringN = String.valueOf(n);
			String[] arrN = stringN.split("");

			
			for(int i = 0; i < arrN.length; i++) {
				int digit = Integer.valueOf(arrN[i]);
				sum += digit;
			}

			n = sum;
		}

		return n;
	}
}

File: golang/digital_root.go
package kata

import (
	"strconv"
	"strings"
)

func DigitalRoot(n int) int {

	for n > 9 {
		str := strconv.Itoa(n)
		arr := strings.Split(str, "")

		intarr := make([]int, 0)

		for _, x := range arr {
			y, err := strconv.Atoi(x)
			intarr = append(intarr, y)

			if err != nil {
				panic(err)
			}
		}

		sum := 0

		for _, el := range intarr {
			sum += el
		}

		n = sum
	}

	return n
}
